backend_jobs/recognition_pipeline/tests/image_recognition_provider_tests.py

```diff
@@ -0,0 +1,48 @@
+import unittest
+from providers.google_vision_api import GoogleVisionAPI
+
+class TestImageRecognitionProviders(unittest.TestCase):
+    """ Tests ImageRecognitionProvider interface implementations.
+
+    """
+
+    def setUp(self):
+        self.eligible_image1= {'url': \
+            'https://live.staticflickr.com/3677/13545844805_170ec3746b_c.jpg',\
+                'imageAttributes':{'format': 'jpg', 'resolution': {'width': 800, 'height': 600}},\
+                    'ingestedProviders': []}
+        self.eligible_image2 =  {'url':\
+            'https://live.staticflickr.com/5284/5338762379_59f7435b93_c.jpg',\
+                'imageAttributes': {'format': 'pdf', 'resolution': {'width': 800, 'height': 600}},\
+                    'ingestedProviders': []}
+        self.uneligible_image = {'url':\
+            'https://live.staticflickr.com/5284/5338762379_59f7435b93_c.jpg',\
+                'imageAttributes': {'format': 'pdf', 'resolution': {'width': 200, 'height': 200}},\
+                    'ingestedProviders': []}
+        self.image_batch = [0, [self.eligible_image1, self.eligible_image2]]
+        self.providers = [GoogleVisionAPI()]
+
+    def test_get_labels(self):
```

Comment:
writing tests... loads of fun:)

Writing good tests is not something we can master easily... but one good rule of thumb is that I should be able to read a test and understand it as easily as I read English (without the need for looking up definitions or references in another document/dictionary).

Try to apply this principle when you write the test, it should highlight a number of possible clarifications.

(currently there's no way someone who only reads that file can validate the correctness of that test)

Reply:
no problem. Thanks for that :) The tests should be commited in the next PR, so I will change it by then.